File: frontend/src/types/phase2.ts
export interface UploadedClothing {
  id: string;
  file: File;
  preview: string;
  analysis_source?: "ai" | "fallback";
  analyzed?: ClothingAnalysis;
}

export interface ClothingAnalysis {
  category: string;
  color: string;
  style: string;
  warmth_level: string;
  weather_suitability: string;
  notes: string;
}

export interface Recommendation {
  day: number;
  date: string;
  outfit_description: string;
  clothing_items: string[];
  weather_match: string;
  confidence: number;
  is_viable: boolean;
  day_warning?: string;
}

export interface WeatherForecast {
  day: number;
  date: string;
  temperature: number;
  condition: string;
  humidity: number;
}

export interface AnalyzeClothesRequest {
  clothing_ids: string[];
}

export interface AnalyzeClothesResponse {
  success: boolean;
  analyses: ClothingAnalysis[];
  message?: string;
}

export interface WeatherForecastRequest {
  location: string;
  days: number;
}

export interface WeatherForecastResponse {
  success: boolean;
  location: string;
  forecast: WeatherForecast[];
  message?: string;
}

export interface RecommendationsGenerateRequest {
  clothing_data: ClothingAnalysis[];
  weather_forecast: WeatherForecast[];
  location: string;
}

export interface RecommendationsGenerateResponse {
  success: boolean;
  recommendations: Recommendation[];
  warnings: string[];
  message?: string;
}

File: backend/app/schemas/phase2.py
from typing import Optional

from pydantic import BaseModel, Field, field_validator

from app.utils.input_validation import sanitize_email, sanitize_location


class ClothingAnalysisSchema(BaseModel):
    item_id: Optional[str] = None
    status: str = "analyzed"
    category: str
    color: str
    style: str
    warmth_level: str
    weather_suitability: str
    gender: str = "Unisex"
    notes: str
    confidence_score: float = 100.0


class AnalyzeClothesRequest(BaseModel):
    clothing_ids: list[str]


class AnalyzeClothesResponse(BaseModel):
    success: bool
    analyses: list[ClothingAnalysisSchema]
    message: Optional[str] = None


class WeatherForecastSchema(BaseModel):
    day: int
    date: str
    temperature: int
    condition: str
    humidity: int
    chance_of_rain: int = 0
    wind_kph: int = 0
    condition_icon: str = ""


class LocationBoundRequest(BaseModel):
    location: str = Field(..., max_length=120)

    @field_validator("location")
    @classmethod
    def validate_location(cls, value: str) -> str:
        return sanitize_location(value)


class WeatherForecastRequest(LocationBoundRequest):
    days: int = Field(default=5, ge=1, le=7)


class WeatherForecastResponse(BaseModel):
    success: bool
    location: str
    forecast: list[WeatherForecastSchema]
    message: Optional[str] = None


class RecommendationSchema(BaseModel):
    day: int
    date: str
    outfit_description: str
    clothing_items: list[str]
    selected_item_ids: list[str] = []
    selected_role_ids: dict[str, Optional[str]] = {}
    weather_match: str
    confidence: float
    recommendation_source: str = "rule-based"
    is_viable: bool = True
    day_warning: Optional[str] = None


class RecommendationsGenerateRequest(LocationBoundRequest):
    clothing_data: list[ClothingAnalysisSchema]
    weather_forecast: list[WeatherForecastSchema]


class RecommendationsGenerateResponse(BaseModel):
    success: bool
    recommendations: list[RecommendationSchema]
    warnings: list[str] = []
    message: Optional[str] = None


class RecommendationRefreshDayRequest(LocationBoundRequest):
    day: int = Field(..., ge=1, le=7)
    clothing_data: list[ClothingAnalysisSchema]
    weather_forecast: list[WeatherForecastSchema]
    current_recommendation: Optional[RecommendationSchema] = None


class RecommendationRefreshDayResponse(BaseModel):
    success: bool
    recommendation: RecommendationSchema
    message: Optional[str] = None


class RecommendationRefreshWeekRequest(LocationBoundRequest):
    clothing_data: list[ClothingAnalysisSchema]
    weather_forecast: list[WeatherForecastSchema]
    current_recommendations: list[RecommendationSchema] = []


class RecommendationRefreshWeekResponse(BaseModel):
    success: bool
    recommendations: list[RecommendationSchema]
    warnings: list[str] = []
    message: Optional[str] = None


class SendPlanWardrobeItemSchema(BaseModel):
    id: str
    file_path: Optional[str] = None
    category: str
    color: str
    gender: str = "Unisex"
    image_data_url: Optional[str] = None


class SendPlanEmailRequest(LocationBoundRequest):
    email: str = Field(..., max_length=254)
    weather_forecast: list[WeatherForecastSchema]
    recommendations: list[RecommendationSchema]
    warnings: list[str] = []
    wardrobe_items: list[SendPlanWardrobeItemSchema] = []

    @field_validator("email")
    @classmethod
    def validate_email(cls, value: str) -> str:
        return sanitize_email(value)


class SendPlanEmailResponse(BaseModel):
    success: bool
    message: Optional[str] = None
